### PR TITLE
doc: networking release notes from the upmerge

### PR DESCRIPTION
**Pull request by rlubos**

### CODE REVIEW

**Review by ru-fu — CHANGES_REQUESTED** [review]
Will add a commit with small fixes.

**Review by ru-fu — COMMENTED** [review]
Added - please check and squash, if okay.

**Comment by rlubos** [comment]
@ru-fu Squashed, thanks.

File: doc/nrf/releases/release-notes-latest.rst [hunk 1]
```diff
@@ -436,17 +462,26 @@ The following list summarizes the most important changes inherited from upstream
       * :option:`CONFIG_OPENTHREAD_MAC_SOFTWARE_RETRANSMIT_ENABLE`
       * :option:`CONFIG_OPENTHREAD_PLATFORM_USEC_TIMER_ENABLE`
       * :option:`CONFIG_OPENTHREAD_CONFIG_PLATFORM_INFO`
+      * :option:`OPENTHREAD_RADIO_LINK_IEEE_802_15_4_ENABLE`
+      * :option:`OPENTHREAD_RADIO_LINK_TREL_ENABLE`
+      * :option:`OPENTHREAD_CSL_SAMPLE_WINDOW`
+      * :option:`OPENTHREAD_CSL_RECEIVE_TIME_AHEAD`
+      * :option:`OPENTHREAD_MAC_SOFTWARE_CSMA_BACKOFF_ENABLE`
```
Comment (ru-fu) [inline doc/nrf/releases/release-notes-latest.rst:469]: Missed that these all need to start with CONFIG_ - sorry. This will fail in the doc build.

Comment (rlubos) [reply doc/nrf/releases/release-notes-latest.rst:469]: Indeed, will fix rightaway.

**Review by ru-fu — APPROVED** [review]

**Comment by ru-fu** [comment]
Should this be merged?

**Comment by rlubos** [comment]
We need more approvals here, @carlescufi?

**Review by carlescufi — APPROVED** [review]

**Pull request merged**